app.js

```diff
@@ -3,14 +3,36 @@ const app = express();
 
 const {
     getEndpoints,
-    getTopics
+    getTopics,
+    getArticlesById,
   } = require('./controllers/news-controller');
 
 app.use(express.json());
 
 app.get('/api', getEndpoints)
 app.get('/api/topics', getTopics)
+app.get('/api/articles/:article_id', getArticlesById)
 
-module.exports = app;
+app.use((err, req, res, next) => {
+    // postgres errors
 
-//
\ No newline at end of file
+    if(err.code == "22P02") {
+      res.status(400).send({ msg: "Bad request" })
+    }
+    else if(err.code == "23502") {
+      res.status(400).send({ msg: 'Bad request'})
+    }
+    else {
+      next(err)
+    }
+  })
+  
+  app.use((err, req, res, next) => {
+    // custom errors
+    if(err.status && err.msg) {
+      res.status(err.status).send({ msg: err.msg })
+    }
+  })
```

Comment:
Good job with these - you can also extract your error handlers out just like a controller